File: app/src/main/java/com/company/moviemania/Adapters/CategoryEachMovieAdapter.java
package com.company.moviemania.Adapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.company.moviemania.Domian.GenresItem;
import com.company.moviemania.R;

import java.util.ArrayList;
import java.util.List;

public class CategoryEachMovieAdapter extends RecyclerView.Adapter<CategoryEachMovieAdapter.ViewHolder> {

    List<String> item;
    Context context;

    public CategoryEachMovieAdapter(List<String> item) {
        this.item = item;
    }

    @NonNull
    @Override
    public CategoryEachMovieAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        context = parent.getContext();
        View inflate = LayoutInflater.from(parent.getContext()).inflate(R.layout.view_category, parent, false);
        return new ViewHolder(inflate);
    }

    @Override
    public void onBindViewHolder(@NonNull CategoryEachMovieAdapter.ViewHolder holder, int position) {
        holder.titleText.setText(item.get(position));
    }

    @Override
    public int getItemCount() {
        return item.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        TextView titleText;


        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            titleText = itemView.findViewById(R.id.TitleTxt);

        }
    }
}

File: app/src/main/java/com/company/moviemania/Adapters/CategoryEachMovieAdapter.kt
package com.company.moviemania.Adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.company.moviemania.R

class CategoryEachMovieAdapter(var item: List<String>) :
    RecyclerView.Adapter<CategoryEachMovieAdapter.ViewHolder>() {
    var context: Context? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        context = parent.context
        val inflate =
            LayoutInflater.from(parent.context).inflate(R.layout.view_category, parent, false)
        return ViewHolder(inflate)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.titleText.text = item[position]
    }

    override fun getItemCount(): Int {
        return item.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var titleText: TextView

        init {
            titleText = itemView.findViewById(R.id.TitleTxt)
        }
    }
}